### ISSUE TITLE
Icons are not properly loading / shown ingame

### ISSUE THREAD

**Issue by RobiSE9722** [lythx/trakman#209]
I have a issue where not all icons are loaded, i am not sure if it is client sided or server sided since some players say everything is loaded and some say it is not.
Any causes?

https://i.gyazo.com/7f2dc63d7ec9746e44969da8ff561b60.jpg

**Comment by wsrvn** [comment]
this has been the main issue with our ui since the very beginning. it is client-side most certainly, as we're (temporarily) using the discord servers (google cdn) to host them. there's no fix that i'm aware of. clearing the game cache would be your best bet.

**Comment by RobiSE9722** [comment]
Thanks for your quick reply. 
Would it not be better to store the icon sets locally instead?
Also do you have a discord server i could join?

**Comment by wsrvn** [comment]
each player would have to load the icons independently of their location.. the only somewhat working solution is letting the player extract all the icons we currently use inside their cache directory manually.

**Comment by wsrvn** [comment]
also no server atm. we do programming, not marketing and promoting (unlike some other very interesting figure in the community). later, when there's a bigger demand for it, perhaps?

**Comment by RobiSE9722** [comment]
What do you mean independently of their location? I was thinking letting them be stored locally on the server but there probably isn't a way to push them to the client that way... didn't think about that. Weird though that they wont properly load when hosted on Discords and google's cnd, i have never had any issues with those outside of this project.
Discord server would be nice just for discussions and troubleshooting. No clue which figure you are talking about.

**Comment by lythx** [comment]
Icon loading stuff is a problem related to the game itself, it can technically be solved by putting the iconpack manually into the cache folder, but that's a lot of effort and the issue usually just goes away after some time. Make sure that you have "locators" option enabled in PeerToPeer menu in game settings, without it the downloads are very slow.

**Comment by wsrvn** [comment]
this issue is now relevant due to discord changing their attachment hotlinking policy. cannot use their cdn anymore, gotta move the icons elsewhere (FAIL)

**Comment by felacek** [comment]
Moved to https://trakman.ptrk.eu/icons

**Comment by sangfroid70** [comment]
Icons are no longer available

**Comment by Skorlok** [comment]
https://trakman.ptrk.eu/icons/FRA.png
https://trakman.ptrk.eu/icons/JPN.png
https://trakman.ptrk.eu/icons/ISR.png
https://trakman.ptrk.eu/icons/PeopleGaming.png
https://trakman.ptrk.eu/icons/Stats.png
https://trakman.ptrk.eu/icons/PaySkip.png

are these available for you ?

**Comment by wsrvn** [comment]
@felacek

**Comment by felacek** [comment]
> Icons are no longer available on https://trakman.ptrk.eu/icons

The listing is 403 forbidden by design, the icons themselves should work?

**Comment by Skorlok** [comment]
the autoindex is disabled by default on NGINX, so the folder return 403, I can personally download individual icons

**Comment by sangfroid70** [comment]
> https://trakman.ptrk.eu/icons/FRA.png https://trakman.ptrk.eu/icons/JPN.png https://trakman.ptrk.eu/icons/ISR.png https://trakman.ptrk.eu/icons/PeopleGaming.png https://trakman.ptrk.eu/icons/Stats.png https://trakman.ptrk.eu/icons/PaySkip.png
> 
> are these available for you ?

Yes available with direct links, but not in game. Strange. Is it possible to have a .zip with icons to put on a local server ? Thank's for your answers anyway

**Comment by wsrvn** [comment]
> Yes available with direct links, but not in game. Strange. Is it possible to have a .zip with icons to put on a local server ?

local server or not, you'd still need to download the pictures through the game for them to appear. the solution we found to be working was to put the icons directly into the game's cache. that ensures each image's availability, but is impossible to influence from our code. the most we could do is to put a warning message for every joining player to download and extract an archive into god knows which directory on their system, which does not sound very convincing.
